resource-types/typescript-example-website-monitor/src/handlers.ts

```diff
@@ -0,0 +1,316 @@
+    readonly status?: string;
+    readonly locations?: string[];
+    readonly slaThreshold?: number;
+}
+
+/**
+ * Resource to be used in AWS CloudFormation to create and manage
+ * New Relic's synthetic monitors of type "ping" (SIMPLE).
+ * See documentation {@link https://docs.newrelic.com/docs/apis/synthetics-rest-api/monitor-examples/manage-synthetics-monitors-rest-api}.
+ */
+class Resource extends BaseResource<ResourceModel> {
+    static readonly DEFAULT_HEADERS = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json'
+    };
+    static readonly DEFAULT_MONITOR_KIND = 'SIMPLE';
+    static readonly DEFAULT_MONITOR_STATUS = 'MUTED';
+    static readonly DEFAULT_MONITOR_LOCATIONS = [
+        'AWS_EU_CENTRAL_1',
+        'AWS_US_WEST_1'
+    ];
+    static readonly DEFAULT_MONITOR_SLA_THRESHOLD = 7.0;
+
+    private async checkResponse(response: Response, logger: LoggerProxy, uid?: string): Promise<any> {
+        if (response.status === 400) {
+            throw new exceptions.AlreadyExists(this.typeName, uid);
+        } else if (response.status === 401) {
+            throw new exceptions.AccessDenied(response.statusText);
+        } else if (response.status === 404) {
+            throw new exceptions.NotFound(this.typeName, uid);
+        } else if (response.status > 400) {
+            throw new exceptions.InternalFailure(
+                `error ${response.status} ${response.statusText}`,
+                HandlerErrorCode.InternalFailure,
+            );
+        }
+        const responseData = await response.text() || '{}';
+        logger.log('HTTP response', responseData);
+        return JSON.parse(responseData);
+    }
+
+    /**
+     * CloudFormation invokes this handler when the resource is initially created
+     * during stack create operations.
+     *
+     * @param session Current AWS session passed through from caller
+     * @param request The request object for the provisioning request passed to the implementor
+     * @param callbackContext Custom context object to allow the passing through of additional
+     * state or metadata between subsequent retries
+     * @param logger Logger to proxy requests to default publishers
+     */
+    @handlerEvent(Action.Create)
+    public async create(
+        session: Optional<SessionProxy>,
+        request: ResourceHandlerRequest<ResourceModel>,
+        callbackContext: CallbackContext,
+        logger: LoggerProxy
+    ): Promise<ProgressEvent> {
+        logger.log('request', request);
+        
+        // It is important that we create a new instance of the model,
+        // because the desired state is immutable.
+        const model = new ResourceModel(request.desiredResourceState);
+        const progress = ProgressEvent.progress<ProgressEvent<ResourceModel>>(model);
+
+        // Id is a read only property, which means that
+        // it cannot be set during creation or update operations.
+        if (model.id) {
```

Comment:
Just a general comment here. I don't believe this should have to be done in the code. The service shouldn't be passing in an `Id` as this is an invalid request based on the fact that its a `readOnlyProperties`.  Can you explain why `Kind`, `Locations`, `Status`, and `SlaThreshold` don't have the same comparisons?

Reply:
Contract tests